File: app/src/main/java/com/example/readative/room/entity/Genre.java
package com.example.readative.room.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Genre {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "genre_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "genre_name")
    public String name;
}

File: app/src/main/java/com/example/readative/persistance/entities/Genre.kt
package com.example.readative.persistance.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "genres",
    indices = [
        Index(value = ["name"], unique = true)
    ]
)
data class Genre (
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "name") val name: String
) : ReadativeEntity